converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/DicomWeb/Service/ChangeFeedRetrieveService.cs

```diff
@@ -39,11 +39,6 @@ public async Task<IReadOnlyList<ChangeFeedEntry>> RetrieveChangeFeedAsync(long o
 
         ChangeFeedEntry[] changeFeedEntries = await result.ToArrayAsync(cancellationToken);
 
-        if (changeFeedEntries?.Any() != null)
-        {
-            return changeFeedEntries;
-        }
-
-        return Array.Empty<ChangeFeedEntry>();
+        return await result.ToArrayAsync(cancellationToken) ?? Array.Empty<ChangeFeedEntry>();
```

Comment:
My comment doesn't concern your change, but I have wondered if the client API shouldn't handle pagination on the users behalf. When you're using the `ChangeFeedRetrieveService`, are you fetching paginated results until the count is 0? If so, perhaps this method (or better yet the underlying client) should return an `IAsyncEnumerable<T>` that automatically handles pagination

Reply:
The way Dicom Cast currently uses this kind of relies on only getting one batch of changes at a time. We process those changes, send them to fhir, and then call ChangeFeedRetrieveService again to see if there's anything else. I'm not sure how getting a larger list of results, or having to wait for multiple pages, could affect the service.

We could change this to handle pagination in theory, and dicom cast might work fine, but I don't want to inflate the work of this too much so that we need to test out changes to dicom cast as well as functions. So I'd prefer to leave it how it is for now